### ISSUE TITLE
Allow parameter values to be provided after translation

### ISSUE THREAD

**Issue by henrikt-ma** [modelica/ModelicaSpecification#2952]
In System Modeler we are currently trying to become strict about parameters with `fixed = true` and neither a declaration equation nor explicit `start`.  Until now, this has only been treated with a warning, and the non-explicit value of `start` has been used.

What we find now is that some models can't be built, even though the parameters don't need to become evaluated.  None of the alternatives for dealing with this appeals to us:
- Providing `start` is just counter-acting the purpose of the rule that requires parameters to be defined.  A warning will be issued, but there is a substantial risk that users won't pay attention.  Also if the typical action to take would be to set `start` even though there's no good default to use, why not simply allow the non-explicit `start` to be used?
- Providing a declaration equation instead of `start` has the same drawbacks, but is even worse as there won't even be a warning when translating the model without providing a sensible parameter value.
- Requiring the parameter when it is time to simulate and creating a one-off model on the fly basically means that the model should be rebuilt each time a different parameter value is used.  This can easily lead to lots of unnecessary rebuilding, and doesn't match intuitive understanding of what it means to simply change the value of a non-structural parameter.

To remedy the situation, one could allow tools to delay the requirement that all parameters with `fixed = true` have values.  For example,
```
model DeferredParameterValue
  parameter Real p;
end DeferredParameterValue;
```
would be legal as a simulation model, but it would be required that no value for `p` is assumed during translation, and tools must require that an explicit value for `p` is provided at initialization.

Can others also see how this would be useful, or am I missing a way to handle this within the constraints of the current specification?

**Comment by HansOlsson** [comment]
> Can others also see how this would be useful, or am I missing a way to handle this within the constraints of the current specification?

I could see how it could be handled within the current specification. You cannot _simulate_ the model without the parameter-values, but you could _translate_ it as a step before that (and then change parameters).

**Comment by henrikt-ma** [comment]
Yes, that's the goal, but would all tools agree that such a model is valid given the current specification text?

**Comment by HansOlsson** [comment]
Separate issues: should only use start-value for parameter if explicitly set.
And only for non-structural (and there can be no cycles for it).

Can then delay setting the parameter from translation until simulation. (No error.)
Agreement. Quality of implementation to be able to translate.

**Comment by henrikt-ma** [comment]
Design meeting:

Alternative design:
```
model DeferredParameterValue
  parameter Real p = break; /* Explicit that there's no value until initialization */
end DeferredParameterValue;
```
